packages/ember-htmlbars/tests/helpers/with_test.js

```diff
@@ -1,262 +1,13 @@
-  view = EmberView.create({
-    template: compile('Admin: {{#with admin as |person|}}{{person.name}}{{/with}} User: {{#with user as |person|}}{{person.name}}{{/with}}'),
-    context: {
-      admin: { name: 'Tom Dale' },
-      user: { name: 'Yehuda Katz' }
-    }
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'Admin: Tom Dale User: Yehuda Katz', 'should be properly scoped');
-});
-
-QUnit.test('should respect `isTruthy` field on a view', function() {
-  view = EmberView.create({
-    template: compile('{{#with view}}True{{else}}False{{/with}}'),
-    isTruthy: true
-  });
-  runAppend(view);
-
-  equal(view.$().text(), 'True');
-
-  run(function() {
-    set(view, 'isTruthy', false);
-  });
-
-  equal(view.$().text(), 'False');
-
-  run(function() {
-    set(view, 'isTruthy', true);
-  });
-
-  equal(view.$().text(), 'True');
-});
-
-QUnit.test('should respect `isTruthy` field on an object', function() {
-  view = EmberView.create({
-    template: compile('{{#with view.foo}}True{{else}}False{{/with}}'),
-    foo: {
-      isTruthy: true
-    }
-  });
-  runAppend(view);
-
-  equal(view.$().text(), 'True');
-
-  run(function() {
-    set(view, 'foo.isTruthy', false);
-  });
-
-  equal(view.$().text(), 'False');
-
-  run(function() {
-    set(view, 'foo.isTruthy', true);
-  });
-
-  equal(view.$().text(), 'True');
-});
-
-QUnit.test('should respect `isTruthy` field on the context object', function() {
-  view = EmberView.create({
-    template: compile('{{#with foo}}True{{else}}False{{/with}}'),
-    context: {
-      foo: {
-        isTruthy: true
-      }
-    }
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'True');
-
-  run(function() {
-    set(view, 'context.foo.isTruthy', false);
-  });
-
-  equal(view.$().text(), 'False');
-
-  run(function() {
-    set(view, 'context.foo.isTruthy', true);
-  });
-
-  equal(view.$().text(), 'True');
-});
-
-QUnit.test('the scoped variable is not available outside the {{with}} block.', function() {
-  view = EmberView.create({
-    template: compile('{{name}}-{{#with other as |name|}}{{name}}{{/with}}-{{name}}'),
-    context: {
-      name: 'Stef',
-      other: 'Yehuda'
-    }
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'Stef-Yehuda-Stef', 'should be properly scoped after updating');
-});
-
-QUnit.test('nested {{with}} blocks shadow the outer scoped variable properly.', function() {
-  view = EmberView.create({
-    template: compile('{{#with first as |ring|}}{{ring}}-{{#with fifth as |ring|}}{{ring}}-{{#with ninth as |ring|}}{{ring}}-{{/with}}{{ring}}-{{/with}}{{ring}}{{/with}}'),
-    context: {
-      first: 'Limbo',
-      fifth: 'Wrath',
-      ninth: 'Treachery'
-    }
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'Limbo-Wrath-Treachery-Wrath-Limbo', 'should be properly scoped after updating');
-});
-
-QUnit.module('Handlebars {{#with keyword as |foo|}}');
-
-QUnit.test('it should support #with view as |foo|', function() {
```

Comment:
Isn't this testing `{{#with this as |that|}}`? (Using the context as the first argument)